code/game/machinery/shieldgen.dm

```diff
@@ -553,3 +553,644 @@
 
 #undef ACTIVE_SETUPFIELDS
 #undef ACTIVE_HASFIELDS
+
+//Modular Shield Generator Start
+/obj/machinery/modular_shield_gen
+	name = "Modular Shield Generator"
+	desc = "A forcefield generator, it seems more stationary than its cousins."
+	icon = 'icons/obj/machines/modular_shield_generator.dmi'
+	icon_state = "gen_recovering_closed"
+	density = TRUE
+	active_power_usage = BASE_MACHINE_ACTIVE_CONSUMPTION * 0.5
+	circuit = /obj/item/circuitboard/machine/modular_shield_gen
+
+	///Doesnt actually control it, just tells us if its running or not, you can control by calling procs activate_shields and deactivate_shields
+	var/active = FALSE
+
+	///Determins if we can turn it on or not, no longer recovering when back to max strength
+	var/recovering = TRUE
+
+	///Determins max health of the shield
+	var/max_strength = 40
+
+	///Current health of shield
+	var/stored_strength = 0 //starts at 0 to prevent rebuild abuse
+
+	///Shield Regeneration when at 100% efficiency
+	var/max_regeneration = 3
+
+	///The regeneration that the shield can support
+	var/current_regeneration
+
+	///Determins the max radius the shield can support
+	var/max_radius = 3
+
+	///Current radius the shield is set to, minimum 3
+	var/radius = 3
+
+	///Determins if we only generate a shield on space turfs or not
+	var/exterior_only = FALSE
+
+	///The list of shields that are ours
+	var/list/deployed_shields = null
+
+	///The list of turfs that are within the shield
+	var/list/inside_shield = null
+
+	///The list of machines that are connected to and boosting us
+	var/list/obj/machinery/modular_shield/module/connected_modules = null
```

Comment:
are you doing this like a lazylist? If you don't know what that is, i would instead do this:

```suggestion
	var/list/obj/machinery/modular_shield/module/connected_modules = list()
```

this is just so we can keep it typed as a list for list-specific procs and variables like `len` and whatever